src/fstext/determinize-star-inl.h

```diff
@@ -552,11 +627,32 @@ template<class Arc> class DeterminizerStar {
               }
               throw std::runtime_error("Non-functional FST: cannot determinize.\n");
             }
-            Weight weight = Plus(iter->second.weight, next_elem.weight);
-            if (! ApproxEqual(weight, iter->second.weight, delta_)) {  // add extra part of weight to queue.
-              iter->second.weight = weight; // Update weight in map.
-              queue.push_back(next_elem);
-              replaced_elems = true;
+            if (iter->second.in_queue) {
+              // since in queue, no check since we're propagating them anyway
+              iter->second.weight_to_process = Plus(iter->second.weight_to_process, next_unprocessed_weight);
+              // probably should take it out of the queue, update the depth
+              // and put it back.
+              // but it seems it's fast enough already ;-)
+            }
+            else {
+              // is in the subset but not in queue (we know its unprcessed_weight == Zero())
```

Comment:
The logic here is what I intended.  You are effectively discarding any weight that does not meet the delta threshold.
What would be better is to *always* accumulate, in iter->second.weight_to_process, the extra weight, but to use the delta threshold to decide whether to put it in the queue (if it isn't already there).